### PR TITLE
Switch heredocs from <<- to <<~

### LINKED ISSUE #232: Use squiggly heredocs everywhere
Squiggly heredocs seem like the "more correct" way to use multiline strings for our use case, and they're supported in all Ruby versions we support, so we should switch to using them everywhere.

### PR DESCRIPTION
**Pull request by JacobEvelyn**
This commit also changes the heredoc closing
signifier to `HEREDOC` everywhere, for clarity.

Closes #232

**Before merging:**

- [ ] ~Copy the table printed at the end of the latest benchmark results into the `README.md` and update this PR~
- [ ] ~If this change merits an update to `CHANGELOG.md`, add an entry following Keep a Changelog [guidelines](https://keepachangelog.com/en/1.0.0/) with [semantic versioning](https://semver.org/)~

### CODE REVIEW

File: benchmarks/benchmarks.rb [hunk 1]
```diff
@@ -63,8 +63,7 @@ def benchmark_name
 # Use metaprogramming to ensure that each class is created in exactly the
 # the same way.
 BENCHMARK_GEMS.each do |benchmark_gem|
-  # rubocop:disable Security/Eval
-  eval <<-CLASS, binding, __FILE__, __LINE__ + 1
+  eval <<~HEREDOC, binding, __FILE__, __LINE__ + 1 # rubocop:disable Security/Eval
```
Comment (JacobEvelyn) [inline benchmarks/benchmarks.rb:66]: I don't love `HEREDOC` but was wondering if we might want something more consistent. Right now we use `CLASS`, `END_OF_METHOD`, and `END_OF_METHODS`. I'm happy to revert this change though and keep them as-is, or switch to something more understandable than `HEREDOC`, like `MULTILINE_STRING`.

Thoughts?

Comment (ms-ati) [reply benchmarks/benchmarks.rb:66]: I am fine with whatever you choose here @JacobEvelyn. Agree that having a consistent usage might be nice.

**Review by ms-ati — APPROVED** [review]
LGTM
